Hunting/Assets/Scripts/GameManager/SunController.cs

```diff
@@ -10,34 +10,94 @@ public class SunController : MonoBehaviour
     private float dayTime = 60;
 
     [SerializeField]
-    private float dimFactor = 3;
+    private float currentTimeOfDay = 0.5f;
+
+    public float secondsInFullDay = 120f;
+    public float timeMultiplier = 0.5f;
+
+    public MeshRenderer cloud;
     
     # endregion
 
     # region Fields
 
-    private float moveAngle = 0;
 
-    private new Light light = null;
+    private new Light sun;
+    private new float sunInitialIntensity;
+
+    private new Material high = null;
+
+    private new float r = 1f;
+    private new float g = 1f;
+    private new float b = 1f;
+
+
+    public System.Collections.Generic.List<Color> colors = new System.Collections.Generic.List<Color>();
 
     # endregion
 
     # region MonoBehaviour Methods
 
     void Start()
     {
-        if (dayTime != 0)
-            moveAngle = 60/dayTime;
         
-        light = GetComponent<Light>();
-        light.intensity = 1;
+        sun = GetComponent<Light>();
+
+        sunInitialIntensity = sun.intensity;
+
+        high = cloud.materials[0];
     }
 
     // Update is called once per frame
     void Update()
     {
-        transform.Rotate(Vector3.up * Time.deltaTime * moveAngle);
-        light.color += (Color.red/(dayTime * dimFactor)) * Time.deltaTime;
+
+        UpdateSun();
+        
+        currentTimeOfDay += (Time.deltaTime / secondsInFullDay) * timeMultiplier;
+ 
+        if (currentTimeOfDay >= 1) {
+            currentTimeOfDay = 0;
+        }
+
+    }
+
+    void UpdateSun() {
+        sun.transform.localRotation = Quaternion.Euler((currentTimeOfDay * 360f) - 90, 170, 0);
+ 
+        float intensityMultiplier = 1;
+        if (currentTimeOfDay >= 0.8f || currentTimeOfDay <= 0.2f)  // Night time

```

Comment:
There are magic numbers in various places in the code. It will be good if you hold them in variables. If not, I am ok with you not changing it, since there's comments explaining what the values mean.